### PR TITLE
[Connector API] Improve create connector endpoints

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

- Since all body params are optional when creating a connector, let's handle empty request body correctly
  - It was reported that people are confused why body always has to be passed (even if empty) 
- Unify create connector API with how index API works when creating new connectors
  - we accept `PUT _connector/{id}` if `id` is not provided doc `_id` is auto-generated
  - for `POST _connector` request `_id` is always auto-generated
  - Unify create connector response type
- Simplify internal connector service logic 
- Add tests empty body, missing `id` path param
- Update docs

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java [hunk 1]
```diff
@@ -127,62 +129,20 @@ public void createConnectorWithDocId(PutConnectorAction.Request request, ActionL
...
-            request.getDescription(),
-            indexName,
-            request.getIsNative(),
-            request.getLanguage(),
-            request.getName(),
-            request.getServiceType()
-        );
 
-        try {
-            isDataIndexNameAlreadyInUse(indexName, null, listener.delegateFailure((l, isIndexNameInUse) -> {
```
Comment (efegurkan) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:167]: Did we lose the `index is already` used case or is it not valid anymore?

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/ConnectorCreateActionResponse.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    private final String id;
+    private final DocWriteResponse.Result result;
+
+    public ConnectorCreateActionResponse(StreamInput in) throws IOException {
+        super(in);
+        this.id = in.readString();
+        this.result = DocWriteResponse.Result.readFrom(in);
+    }
+
+    public ConnectorCreateActionResponse(String id, DocWriteResponse.Result result) {
```
Comment (efegurkan) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/ConnectorCreateActionResponse.java:32]: This is a question for me to understand rather, when this is used and why?

**Review by efegurkan — APPROVED** [review]
Looks good! 

A few questions rather for myself to understand.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java (hunk 1 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:167]: No, we used to have `createConnectorWithAutoGeneratedId` and `createConnectorWithDocId`. I unified this into a single `createConnector()` that takes connectorID as an arg (if not present we autogenerate). We still have the check `index is already used` in place

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/ConnectorCreateActionResponse.java (hunk 2 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/ConnectorCreateActionResponse.java:32]: This is a Java class that represents a response object, the toXContent method allows this to be converted into JSON representation

**Pull request merged**